cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -159,7 +165,7 @@ public void startPreview(PreviewRequest<?> previewRequest) throws Exception {
                                             artifactSummary, config, NOOP_PROGRAM_TERMINATOR, null,
                                             request.canUpdateSchedules());
     } catch (Exception e) {
-      this.status = new PreviewStatus(PreviewStatus.Status.DEPLOY_FAILED, new BasicThrowable(e), null, null);
+      setStatus(new PreviewStatus(PreviewStatus.Status.DEPLOY_FAILED, new BasicThrowable(e), null, null));
```

Comment:
I can't comment on unchanged code. But in line 183, 

- give the Timer a name based on the preview id so that the thread it creates is named.
- Suggest to just new the `Timer` on the startUp and always close it in shutDown.